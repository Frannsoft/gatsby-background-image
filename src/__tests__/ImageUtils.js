import { render } from 'react-testing-library'
import 'react-testing-library/cleanup-after-each'
import React from 'react'
import { fixedShapeMock, fluidShapeMock } from './mocks/Various.mock'
import {
  noscriptImg,
  inImageCache,
  activateCacheForImage,
  resetImageCache,
  createPictureRef,
  switchImageSettings,
  imagePropsChanged,
} from '../ImageUtils'

global.console.debug = jest.fn()

afterEach(resetImageCache)

const fixedMock = {
  fixed: fixedShapeMock,
}

const fluidMock = {
  fluid: fluidShapeMock,
}

describe(`createImageToLoad()`, () => {
  it(`should return null on ssr or empty fluid / fixed prop`, () => {
    const emptyImageRef = createPictureRef({})
    expect(emptyImageRef).toBeNull()
  })

  it(`should return null on ssr or empty fluid / fixed prop & wrong type for onLoad`, () => {
    const emptyImageRef = createPictureRef({}, {})
    expect(emptyImageRef).toBeNull()
  })
})

describe(`createImageToLoad() with crossOrigin`, () => {
  it(`should set crossOrigin`, () => {
    const emptyImageRef = createPictureRef({
      fluid: fluidShapeMock,
      crossOrigin: `anonymous`,
    })
    expect(emptyImageRef).toMatchInlineSnapshot(`
      <img
        crossorigin="anonymous"
        src="test_fluid_image.jpg"
        srcset="some srcSet"
      />
    `)
  })
})

describe(`inImageCache() / activateCacheForImage()`, () => {
  it(`should return false for initial load (fixed)`, () => {
    fixedMock.src = `test.jpg`
    const inCache = inImageCache(fixedMock)
    expect(inCache).toBeFalsy()
  })

  it(`shouldn't activate cache without fixed / fluid`, () => {
    activateCacheForImage()
    const inCache = inImageCache()
    expect(inCache).toBeFalsy()
  })

  it(`should return img from cache after activateCacheForImage() (fixed)`, () => {
    activateCacheForImage(fixedMock)
    const inCache = inImageCache(fixedMock)
    expect(inCache).toBeTruthy()
  })

  it(`inImageCache() should return false for initial load (fluid)`, () => {
    const inCache = inImageCache(fluidMock)
    expect(inCache).toBeFalsy()
  })

  it(`should return img from cache after activateCacheForImage() (fluid)`, () => {
    activateCacheForImage(fluidMock)
    const inCache = inImageCache(fluidMock)
    expect(inCache).toBeTruthy()
  })

  it(`should return false without mock`, () => {
    const inCache = inImageCache()
    expect(inCache).toBeFalsy()
  })
})

describe(`inImageCache() / activateCacheForImage() / resetImageCache()`, () => {
  it(`should reset imageCache after caching`, () => {
    activateCacheForImage(fluidMock)
    resetImageCache()
    const inCache = inImageCache(fluidMock)
    expect(inCache).toBeFalsy()
  })
})

describe(`noscriptImg()`, () => {
  it(`should return default noscriptImg on {}`, () => {
    const { container } = render(noscriptImg({}))
    expect(container).toMatchSnapshot()
  })

  it(`should return noscriptImg with opacity & transitionDelay & crossOrigin`, () => {
    const dummyProps = {
      opacity: 0.99,
      transitionDelay: 100,
      crossOrigin: `anonymous`
    }
    const { container } = render(noscriptImg(dummyProps))
    expect(container).toMatchSnapshot()
  })
})

describe(`switchImageSettings()`, () => {
  const mockImageRef = {
    src: `test.jpg`,
    currentSrc: `test.webp`,
  }
  const state = {
    isVisible: false,
    isLoaded: false,
  }
  it(`should return settings from fluid with empty bgImage`, () => {
    const createdSettings = switchImageSettings({
      image: fluidShapeMock,
      bgImage: ``,
      imageRef: mockImageRef,
      state,
    })
    expect(createdSettings).toMatchSnapshot()
  })

  it(`should return settings from fixed with set bgImage`, () => {
    const createdSettings = switchImageSettings({
      image: fixedShapeMock,
      bgImage: `string_of_base64`,
      imageRef: mockImageRef,
      state,
    })
    expect(createdSettings).toMatchSnapshot()
    expect(createdSettings.lastImage).toEqual(`string_of_base64`)
  })

  it(`should return settings from fixed with set bgImage`, () => {
    const createdSettings = switchImageSettings({
      image: fluidShapeMock,
      bgImage: `string_of_base64`,
      imageRef: mockImageRef,
      state,
    })
    expect(createdSettings).toMatchSnapshot()
  })

  it(`should return settings from fluid with set bgImage without base64`, () => {
    const fluid = fluidShapeMock
    delete fluid.base64
    const createdSettings = switchImageSettings({
      image: fluid,
      bgImage: ``,
      imageRef: mockImageRef,
      state,
    })
    expect(createdSettings).toMatchSnapshot()
  })

  it(`should return settings from fluid with set bgImage and tracedSVG`, () => {
    const fluid = {
      ...fluidShapeMock,
      tracedSVG: 'test_tracedSVG.svg',
    }
    delete fluid.base64
    const createdSettings = switchImageSettings({
      image: fluid,
      bgImage: `test_tracedSVG.svg`,
      imageRef: mockImageRef,
      state,
    })
    expect(createdSettings).toMatchSnapshot()
  })

  it(`should return settings from fluid with set bgImage and base64`, () => {
    const mockRef = mockImageRef
    delete mockRef.src
    state.isVisible = true
    state.isLoaded = true
    const createdSettings = switchImageSettings({
      image: fluidShapeMock,
      bgImage: `string_of_base64`,
      imageRef: mockRef,
      state,
    })
    expect(createdSettings).toMatchSnapshot()
  })
})

describe(`imagePropsChanged()`, () => {
  it(`should return false for same fixed prop`, () => {
    const changedFixed = imagePropsChanged(fixedMock, fixedMock)
    expect(changedFixed).toBeFalsy()
  })

  it(`should return false for same fluid prop`, () => {
    const changedFluid = imagePropsChanged(fluidMock, fluidMock)
    expect(changedFluid).toBeFalsy()
  })

  it(`should return true from fluid to fixed`, () => {
    const changedFluidToFixed = imagePropsChanged(fluidMock, fixedMock)
    expect(changedFluidToFixed).toBeTruthy()
  })

  it(`should return true from fixed to fluid`, () => {
    const changedFixedToFluid = imagePropsChanged(fixedMock, fluidMock)
    expect(changedFixedToFluid).toBeTruthy()
  })
})
