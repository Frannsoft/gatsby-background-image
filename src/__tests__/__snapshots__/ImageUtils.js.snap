// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundImage /> should call critical fixed images 1`] = `
<div>
  <div
    class="fixedImage gatsby-background-image-test gatsby-image-wrapper"
    style="position: relative; overflow: hidden; display: inline; width: 100px; height: 100px; opacity: 0.99; background-size: 'contain';"
    title="Title for the image"
  >
    <style>
      
          .fixedImage.:before,
.gatsby-background-image-test:before,
          .fixedImage.:after,
.gatsby-background-image-test:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition: none;
            background-position: 'center';
background-repeat: 'repeat-y';
background-size: 'contain';

          }
          .fixedImage.:before,
.gatsby-background-image-test:before {
            z-index: -100;
            
            background-image: url(test_fixed_image.jpg);
            background-color: lightgray;
            opacity: 1; 
          }
          .fixedImage.:after,
.gatsby-background-image-test:after {
            z-index: -101;
            background-image: url(test_fixed_image.jpg);
            
            background-color: lightgray;
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" /&gt;&lt;img width="100" height="100" srcset="some srcSet" src="test_fixed_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`<BackgroundImage /> should call critical fluid images 1`] = `
<div>
  <div
    class="fixedImage gatsby-background-image-test gatsby-image-wrapper"
    style="position: relative; overflow: hidden; opacity: 0.99; display: inline; background-size: 'contain';"
    title="Title for the image"
  >
    <style>
      
          .fixedImage.:before,
.gatsby-background-image-test:before,
          .fixedImage.:after,
.gatsby-background-image-test:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: 'center';
background-repeat: 'repeat-y';
background-size: 'contain';

          }
          .fixedImage.:before,
.gatsby-background-image-test:before {
            z-index: -100;
            
            
            background-color: lightgray;
            opacity: 0; 
          }
          .fixedImage.:after,
.gatsby-background-image-test:after {
            z-index: -101;
            
            background-image: url(test_fluid_image.jpg);
            background-color: lightgray;
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test_fluid_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`<BackgroundImage /> should change style.display from 'inherit' to 'inline-block' 1`] = `
<div>
  <div
    class="gatsby-background-image-fwatluf gatsby-image-wrapper"
    style="position: relative; overflow: hidden; width: 100px; height: 100px; opacity: 0.99; background-position: center; background-repeat: no-repeat; background-size: cover;"
    title=""
  >
    <style>
      
          .gatsby-background-image-fwatluf:before,
          .gatsby-background-image-fwatluf:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .gatsby-background-image-fwatluf:before {
            z-index: -100;
            
            background-image: url(test_fixed_image.jpg);
            
            opacity: 1; 
          }
          .gatsby-background-image-fwatluf:after {
            z-index: -101;
            background-image: url(string_of_base64);
            
            
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" /&gt;&lt;img width="100" height="100" srcset="some srcSet" src="test_fixed_image.jpg" alt="" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      testempty
    </h1>
  </div>
</div>
`;

exports[`<BackgroundImage /> should create style tag at fixed with pseudo-elements 1`] = `
<style>
  
          .fixedImage.test:before,
.gatsby-background-image-test:before,
          .fixedImage.test:after,
.gatsby-background-image-test:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: 'center';
background-repeat: 'repeat-y';
background-size: 'contain';

          }
          .fixedImage.test:before,
.gatsby-background-image-test:before {
            z-index: -100;
            
            background-image: url(test_fixed_image.jpg);
            background-color: lightgray;
            opacity: 1; 
          }
          .fixedImage.test:after,
.gatsby-background-image-test:after {
            z-index: -101;
            background-image: url(string_of_base64);
            
            background-color: lightgray;
          }
        
</style>
`;

exports[`<BackgroundImage /> should create style tag with pseudo-elements 1`] = `
<style>
  
          .fixedImage.test:before,
.gatsby-background-image-test:before,
          .fixedImage.test:after,
.gatsby-background-image-test:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: 'center';
background-repeat: 'repeat-y';
background-size: 'contain';

          }
          .fixedImage.test:before,
.gatsby-background-image-test:before {
            z-index: -100;
            
            background-image: url(test_fixed_image.jpg);
            background-color: lightgray;
            opacity: 1; 
          }
          .fixedImage.test:after,
.gatsby-background-image-test:after {
            z-index: -101;
            background-image: url(string_of_base64);
            
            background-color: lightgray;
          }
        
</style>
`;

exports[`<BackgroundImage /> should render fixed size images 1`] = `
<div>
  <div
    class="fixedImage gatsby-background-image-test gatsby-image-wrapper"
    style="position: relative; overflow: hidden; display: inline; width: 100px; height: 100px; opacity: 0.99; background-position: center; background-repeat: no-repeat; background-size: cover;"
    title="Title for the image"
  >
    <style>
      
          .fixedImage.:before,
.gatsby-background-image-test:before,
          .fixedImage.:after,
.gatsby-background-image-test:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .fixedImage.:before,
.gatsby-background-image-test:before {
            z-index: -100;
            
            background-image: url(test_fixed_image.jpg);
            background-color: lightgray;
            opacity: 1; 
          }
          .fixedImage.:after,
.gatsby-background-image-test:after {
            z-index: -101;
            background-image: url(string_of_base64);
            
            background-color: lightgray;
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" /&gt;&lt;img width="100" height="100" srcset="some srcSet" src="test_fixed_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`<BackgroundImage /> should render fluid images 1`] = `
<div>
  <div
    class="fixedImage gatsby-background-image-test gatsby-image-wrapper"
    style="position: relative; overflow: hidden; opacity: 0.99; display: inline; background-size: 'contain';"
    title="Title for the image"
  >
    <style>
      
          .fixedImage.:before,
.gatsby-background-image-test:before,
          .fixedImage.:after,
.gatsby-background-image-test:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: 'center';
background-repeat: 'repeat-y';
background-size: 'contain';

          }
          .fixedImage.:before,
.gatsby-background-image-test:before {
            z-index: -100;
            
            background-image: url(test_fluid_image.jpg);
            background-color: lightgray;
            opacity: 1; 
          }
          .fixedImage.:after,
.gatsby-background-image-test:after {
            z-index: -101;
            background-image: url(string_of_base64);
            
            background-color: lightgray;
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test_fluid_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`<BackgroundImage /> should return null without fluid / fixed 1`] = `<div />`;

exports[`<BackgroundImage /> should work with BackgroundColor 1`] = `
<div>
  <div
    class="gatsby-background-image-fwatluf gatsby-image-wrapper"
    style="position: relative; overflow: hidden; opacity: 0.99; display: inline; background-position: center; background-repeat: no-repeat; background-size: cover;"
    title="Title for the image"
  >
    <style>
      
          .gatsby-background-image-fwatluf:before,
          .gatsby-background-image-fwatluf:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .gatsby-background-image-fwatluf:before {
            z-index: -100;
            
            background-image: url(test_fluid_image.jpg);
            background-color: #fff;
            opacity: 1; 
          }
          .gatsby-background-image-fwatluf:after {
            z-index: -101;
            background-image: url(string_of_base64);
            
            background-color: #fff;
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test_fluid_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`<BackgroundImage /> should work with another external class 1`] = `
<div>
  <div
    class="fixedImage testgatsby-background-image-test gatsby-image-wrapper"
    style="position: relative; overflow: hidden; opacity: 0.99; display: inline; background-size: 'contain';"
    title="Title for the image"
  >
    <style>
      
          .fixedImage.test:before,
.gatsby-background-image-test:before,
          .fixedImage.test:after,
.gatsby-background-image-test:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: 'center';
background-repeat: 'repeat-y';
background-size: 'contain';

          }
          .fixedImage.test:before,
.gatsby-background-image-test:before {
            z-index: -100;
            
            
            background-color: lightgray;
            opacity: 0; 
          }
          .fixedImage.test:after,
.gatsby-background-image-test:after {
            z-index: -101;
            
            background-image: url(test_fluid_image.jpg);
            background-color: lightgray;
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test_fluid_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`<BackgroundImage /> should work with fadeIn 1`] = `
<div>
  <div
    class="gatsby-background-image-fwatluf gatsby-image-wrapper"
    style="position: relative; overflow: hidden; opacity: 0.99; display: inline; background-position: center; background-repeat: no-repeat; background-size: cover;"
    title="Title for the image"
  >
    <style>
      
          .gatsby-background-image-fwatluf:before,
          .gatsby-background-image-fwatluf:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .gatsby-background-image-fwatluf:before {
            z-index: -100;
            
            background-image: url(test_fluid_image.jpg);
            background-color: lightgray;
            opacity: 1; 
          }
          .gatsby-background-image-fwatluf:after {
            z-index: -101;
            background-image: url(string_of_base64);
            
            background-color: lightgray;
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test_fluid_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`<BackgroundImage /> should work with only classId 1`] = `
<div>
  <div
    class="gatsby-background-image-fwatluf gatsby-image-wrapper"
    style="position: relative; overflow: hidden; opacity: 0.99; display: inline; background-position: center; background-repeat: no-repeat; background-size: cover;"
    title="Title for the image"
  >
    <style>
      
          .gatsby-background-image-fwatluf:before,
          .gatsby-background-image-fwatluf:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .gatsby-background-image-fwatluf:before {
            z-index: -100;
            
            background-image: url(test_fluid_image.jpg);
            
            opacity: 1; 
          }
          .gatsby-background-image-fwatluf:after {
            z-index: -101;
            background-image: url(string_of_base64);
            
            
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test_fluid_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`<BackgroundImage /> should work without classId but className 1`] = `
<div>
  <div
    class="fixedImage test gatsby-image-wrapper"
    style="position: relative; overflow: hidden; opacity: 0.99; display: inline; background-size: 'contain';"
    title="Title for the image"
  >
    <style>
      
          .fixedImage.test:before,
          .fixedImage.test:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: 'center';
background-repeat: 'repeat-y';
background-size: 'contain';

          }
          .fixedImage.test:before {
            z-index: -100;
            
            background-image: url(test_fluid_image.jpg);
            
            opacity: 1; 
          }
          .fixedImage.test:after {
            z-index: -101;
            background-image: url(string_of_base64);
            
            
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test_fluid_image.jpg" alt="Alt text for the image" title="Title for the image" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      test
    </h1>
  </div>
</div>
`;

exports[`<BackgroundImage /> should work without external class 1`] = `
<div>
  <div
    class="gatsby-background-image-fwatluf gatsby-image-wrapper"
    style="position: relative; overflow: hidden; opacity: 0.99; background-position: center; background-repeat: no-repeat; background-size: cover;"
    title=""
  >
    <style>
      
          .gatsby-background-image-fwatluf:before,
          .gatsby-background-image-fwatluf:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            transition-delay: 0.25s;
            transition: opacity 0.5s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .gatsby-background-image-fwatluf:before {
            z-index: -100;
            
            background-image: url(test_fluid_image.jpg);
            
            opacity: 1; 
          }
          .gatsby-background-image-fwatluf:after {
            z-index: -101;
            background-image: url(string_of_base64);
            
            
          }
        
    </style>
    <noscript>
      &lt;picture&gt;&lt;source type='image/webp' srcset="some srcSetWebp" sizes="(max-width: 600px) 100vw, 600px" /&gt;&lt;img sizes="(max-width: 600px) 100vw, 600px" srcset="some srcSet" src="test_fluid_image.jpg" alt="" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
    </noscript>
    <h1>
      testempty
    </h1>
  </div>
</div>
`;

exports[`noscriptImg() should return default noscriptImg on {} 1`] = `
<div>
  &lt;picture&gt;&lt;img src="" alt="" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
</div>
`;

exports[`noscriptImg() should return noscriptImg with opacity & transitionDelay & crossOrigin 1`] = `
<div>
  &lt;picture&gt;&lt;img src="" alt="" crossorigin="anonymous" style="position:absolute;top:0;left:0;z-index:-1;transition:opacity 0.5s;transition-delay:100;opacity:0.99;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
</div>
`;

exports[`switchImageSettings() should return settings from fixed with set bgImage 1`] = `
Object {
  "afterOpacity": 1,
  "bgImage": "",
  "lastImage": "string_of_base64",
  "nextImage": "string_of_base64",
  "noBase64": false,
}
`;

exports[`switchImageSettings() should return settings from fixed with set bgImage 2`] = `
Object {
  "afterOpacity": 1,
  "bgImage": "",
  "lastImage": "string_of_base64",
  "nextImage": "string_of_base64",
  "noBase64": false,
}
`;

exports[`switchImageSettings() should return settings from fluid with empty bgImage 1`] = `
Object {
  "afterOpacity": 0,
  "bgImage": "string_of_base64",
  "lastImage": "",
  "nextImage": "",
  "noBase64": false,
}
`;

exports[`switchImageSettings() should return settings from fluid with set bgImage and base64 1`] = `
Object {
  "afterOpacity": 1,
  "bgImage": "",
  "lastImage": "string_of_base64",
  "nextImage": "test_fluid_image.jpg",
  "noBase64": true,
}
`;

exports[`switchImageSettings() should return settings from fluid with set bgImage and tracedSVG 1`] = `
Object {
  "afterOpacity": 1,
  "bgImage": "",
  "lastImage": "test_tracedSVG.svg",
  "nextImage": "\\"test_tracedSVG.svg\\"",
  "noBase64": true,
}
`;

exports[`switchImageSettings() should return settings from fluid with set bgImage without base64 1`] = `
Object {
  "afterOpacity": 0,
  "bgImage": "",
  "lastImage": "string_of_base64",
  "nextImage": "",
  "noBase64": true,
}
`;
